modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/frontend/taglib/servlet/taglib/CommerceDiscountDetailsScreenNavigationEntry.java

```diff
@@ -0,0 +1,164 @@
+import com.liferay.portal.kernel.security.permission.resource.ModelResourcePermission;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.io.IOException;
+
+import java.util.Locale;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Pedro Victor Silvestre
+ */
+@Component(
+	property = "screen.navigation.entry.order:Integer=10",
+	service = ScreenNavigationEntry.class
+)
+public class CommerceDiscountDetailsScreenNavigationEntry
+	implements ScreenNavigationEntry<CommerceDiscount> {
+
+	@Override
+	public String getCategoryKey() {
+		return _commerceDiscountDetailsScreenNavigationCategory.
+			getCategoryKey();
+	}
+
+	@Override
+	public String getEntryKey() {
+		return getCategoryKey();
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return _commerceDiscountDetailsScreenNavigationCategory.getLabel(
+			locale);
+	}
+
+	@Override
+	public String getScreenNavigationKey() {
+		return _commerceDiscountDetailsScreenNavigationCategory.
+			getScreenNavigationKey();
+	}
+
+	@Override
+	public boolean isVisible(User user, CommerceDiscount commerceDiscount) {
+		if (commerceDiscount == null) {
+			return false;
+		}
+
+		boolean hasPermission = false;
+
+		try {
+			hasPermission = _commerceDiscountModelResourcePermission.contains(
+				PermissionThreadLocal.getPermissionChecker(),
+				commerceDiscount.getCommerceDiscountId(), ActionKeys.UPDATE);
+		}
+		catch (Exception exception) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(exception);
+			}
+		}
+
+		return hasPermission;
+	}
+
+	@Override
+	public void render(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse)
+		throws IOException {
+
+		CommerceDiscountDisplayContext commerceDiscountDisplayContext =
+			new CommerceDiscountDisplayContext(
+				_commerceCurrencyLocalService,
+				_commerceDiscountModelResourcePermission,
+				_commerceDiscountService, _commerceDiscountRuleService,
+				_commerceDiscountRuleTypeRegistry,
+				_commerceDiscountTargetRegistry, _percentageFormatter,
+				httpServletRequest, _portal);
+
+		httpServletRequest.setAttribute(
+			WebKeys.PORTLET_DISPLAY_CONTEXT, commerceDiscountDisplayContext);
+
+		_jspRenderer.renderJSP(
+			httpServletRequest, httpServletResponse,
+			"/commerce_discounts/details.jsp");
+	}
+
+	private static final Log _log = LogFactoryUtil.getLog(
+		CommerceDiscountDetailsScreenNavigationEntry.class);
+
+	@Reference
+	private CommerceCurrencyLocalService _commerceCurrencyLocalService;
+
+	@Reference(
+		target = "(component.name=com.liferay.commerce.pricing.web.internal.frontend.taglib.servlet.taglib.CommerceDiscountDetailsScreenNavigationCategory)"
+	)
+	private ScreenNavigationCategory
```

Comment:
@daiazius @rafaprax no need to add reference to category, use the constants directly. This is to make it consistent with current ScreenNavigationEntry, also when rendering on page, you will get category first, then find its entries, no need to have this kind of dependecy.